assets/src/scripts/monitoring-location/components/hydrograph/download-links.js

```diff
@@ -87,43 +58,53 @@ export const renderDownloadLinks = function(elem, store, siteno) {
                 }
         };
 
-        if (hasIVData) {
+        if (hasIVData && primaryIVData && Object.keys(primaryIVData.values) > 0) {
             listOfDownloadLinks.append('li')
                 .call(createDataDownloadLink, {
                     displayText: 'Current IV data',
-                    url: createUrlForDownloadLinks(currentIVDateRange, queryInformation, parameterCode, 'current'),
+                    url: getServiceURL({
+                        monitoringLocations: monitoringLocations,
+                        parameterCode: inputs.parameterCode,
+                        startTime: startDT,
+                        endTime: endDT,
+                        format: 'rdb'
+                    }),
                     gaEventAction: 'downloadLinkCurrent',
                     tooltipText: 'Monitoring location data as shown on graph'
                 });
         }
 
-        if (hasIVData && showIVTimeSeries.compare) {
+        if (inputs.loadCompare && compareIVData && Object.keys(compareIVData.values) > 0) {
             listOfDownloadLinks.append('li')
                 .call(createDataDownloadLink, {
                     displayText: 'Compare IV data',
-                    url: createUrlForDownloadLinks(currentIVDateRange, queryInformation, parameterCode, 'compare'),
+                    url: getServiceURL({
+                        monitoringLocations: monitoringLocations,
+                        parameterCode: inputs.parameterCode,
+                        startTime: DateTime.fromMillis(priorYearTimeRange.start).toISO(),
+                        endTime: DateTime.fromMillis(priorYearTimeRange.end).toISO(),
+                        format: 'rdb'
+                    }),
                     gaEventAction: 'downloadLinkCompare',
                     tooltipText: 'Data from last year with the same duration as in graph'
                 });
         }
 
-        if (showIVTimeSeries.median && medianData) {
+        if (inputs.loadMedian && medianStatisticsData && Object.keys(medianStatisticsData) > 0) {
             listOfDownloadLinks.append('li')
                 .call(createDataDownloadLink, {
                     displayText: 'Median data',
-                    url: `${config.SERVICE_ROOT}/stat/?format=rdb&sites=${siteno}&statReportType=daily&statTypeCd=median&parameterCd=${parameterCode}`,
+                    url: `${config.SERVICE_ROOT}/stat/?sites=${siteno}&statReportType=daily&statTypeCd=median&parameterCd=${inputs.parameterCode}&format=rdb`,
```

Comment:
I think you should follow the same pattern as with the IV data and put a get*Url function in statistics-data and groundwater-levels. We know for sure at some point these URLs will change when they are reimplemented. Putting the code in one place will help keep everything easy to change.

Reply:
I hesitated on moving these because they only get used once, but following the pattern makes sense. Will move.